db/crud.go

```diff
@@ -1922,6 +1922,10 @@ func (db *DatabaseCollectionWithUser) updateAndReturnDoc(ctx context.Context, do
 			if err != nil {
 				return
 			}
+			// If importing and the sync function has modified the expiry, allow sgbucket.MutateInOptions to modify the expiry
+			if allowImport && syncFuncExpiry != nil && *syncFuncExpiry != expiry {
```

Comment:
What case does the allowImport check here protect us from?  (allowImport isn't the same as "this is an import mutation")